common/reactive/src/test/java/io/helidon/common/reactive/MultiFlatMapOptionalTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.common.reactive;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+
+public class MultiFlatMapOptionalTest {
```

Comment:
I think this should also have tests when the originating stream is not a stream of optionals, such as:
```
     Multi.range(1, 5)
                .flatMapOptional(i -> (i % 2) == 0 ? Optional.of(i) : Optional.<Integer>empty()))
                .subscribe(subscriber);
```
Which should work as well (and should cut the middle man of converting the actual value to optional).
This is valid for all tests I think

Reply:
Not sure I follow, you mean something like this?
```java
        Multi.range(1, 5)
                .flatMapOptional(Optional::of)
                .subscribe(subscriber);
```

Reply:
your `flatMapOptional(Function.identity())` is redundant

Reply:
fixed